### PR TITLE
[vmmDomP] Added classes to configure childern and relationship for vmmDomP class #175

### PR DESCRIPTION
**Pull request by anvitha-jain**
Solves issue #175

### CODE REVIEW

File: client/vmmCtrlrP_service.go [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+	return list, err
+}
+
+func (sm *ServiceManager) CreateRelationvmmRsMcastAddrNs(parentDn, tnFvnsMcastAddrInstPName string) error {
+	dn := fmt.Sprintf("%s/rsmcastAddrNs", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s",
+				"tnFvnsMcastAddrInstPName": "%s"
```
Comment (AshuSoni-crest) [inline client/vmmCtrlrP_service.go:62]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/vmmCtrlrP_service.go [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+	return sm.DeleteByDn(dn, "vmmRsMcastAddrNs")
+}
+
+func (sm *ServiceManager) CreateRelationvmmRsAcc(parentDn, vmmUsrAccPDn string) error {
+	dn := fmt.Sprintf("%s/rsacc", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s",
+				"tDn": "%s"
```
Comment (AshuSoni-crest) [inline client/vmmCtrlrP_service.go:97]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/vmmCtrlrP_service.go [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+func (sm *ServiceManager) DeleteRelationvmmRsAcc(parentDn string) error {
+	dn := fmt.Sprintf("%s/rsacc", parentDn)
+	return sm.DeleteByDn(dn, "vmmRsAcc")
+}
+func (sm *ServiceManager) CreateRelationvmmRsVmmCtrlrP(parentDn, tDn string) error {
+	dn := fmt.Sprintf("%s/rsvmmCtrlrP-[%s]", parentDn, tDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"
```
Comment (AshuSoni-crest) [inline client/vmmCtrlrP_service.go:130]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/vmmCtrlrP_service.go [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+	dn := fmt.Sprintf("%s/rsvmmCtrlrP-[%s]", parentDn, tDn)
+	return sm.DeleteByDn(dn, "vmmRsVmmCtrlrP")
+}
+func (sm *ServiceManager) CreateRelationvmmRsVxlanNs(parentDn, tnFvnsVxlanInstPName string) error {
+	dn := fmt.Sprintf("%s/rsvxlanNs", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s","tnFvnsVxlanInstPName": "%s"
+
```
Comment (AshuSoni-crest) [inline client/vmmCtrlrP_service.go:164]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/vmmCtrlrP_service.go [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+	dn := fmt.Sprintf("%s/rsvxlanNs", parentDn)
+	return sm.DeleteByDn(dn, "vmmRsVxlanNs")
+}
+func (sm *ServiceManager) CreateRelationvmmRsCtrlrPMonPol(parentDn, tnMonInfraPolName string) error {
+	dn := fmt.Sprintf("%s/rsctrlrPMonPol", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s","tnMonInfraPolName": "%s"
+
```
Comment (AshuSoni-crest) [inline client/vmmCtrlrP_service.go:198]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/vmmCtrlrP_service.go [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+	dn := fmt.Sprintf("%s/rsctrlrPMonPol", parentDn)
+	return sm.DeleteByDn(dn, "vmmRsCtrlrPMonPol")
+}
+func (sm *ServiceManager) CreateRelationvmmRsMgmtEPg(parentDn, tnFvEPgName string) error {
+	dn := fmt.Sprintf("%s/rsmgmtEPg", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s","tnFvEPgName": "%s"
+
```
Comment (AshuSoni-crest) [inline client/vmmCtrlrP_service.go:232]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/vmmUsrAccP_service.go [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+)
+
+
+
+
+
+
+
+
+
```
Comment (AshuSoni-crest) [inline client/vmmUsrAccP_service.go:17]: Remove these extra lines

File: client/vmmVSwitchPolicyCont_service.go [hunk 8]
```diff
@@ -0,0 +1,331 @@
...
+
+func (sm *ServiceManager) CreateRelationvmmRsVswitchExporterPol(parentDn, tDn, activeFlowTimeout, idleFlowTimeout, samplingRate string) (string, error) {
+	dn := fmt.Sprintf("%s/rsvswitchExporterPol-[%s]", parentDn, tDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s",
+				"activeFlowTimeOut": "%s",
+				"idleFlowTimeOut": "%s",
+				"samplingRate": "%s"
```
Comment (AshuSoni-crest) [inline client/vmmVSwitchPolicyCont_service.go:64]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/vmmVSwitchPolicyCont_service.go [hunk 9]
```diff
@@ -0,0 +1,331 @@
...
+	return sm.DeleteByDn(dn, "vmmRsVswitchExporterPol")
+}
+
+func (sm *ServiceManager) CreateRelationvmmRsVswitchOverrideFwPol(parentDn, tDn string) error {
+	dn := fmt.Sprintf("%s/rsvswitchOverrideFwPol", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s",
+				"tDn": "%s"
```
Comment (AshuSoni-crest) [inline client/vmmVSwitchPolicyCont_service.go:99]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/vmmVSwitchPolicyCont_service.go [hunk 10]
```diff
@@ -0,0 +1,331 @@
...
+	dn := fmt.Sprintf("%s/rsvswitchOverrideFwPol", parentDn)
+	return sm.DeleteByDn(dn, "vmmRsVswitchOverrideFwPol")
+}
+func (sm *ServiceManager) CreateRelationvmmRsVswitchOverrideStpPol(parentDn, StpIfPolName string) error {
+	dn := fmt.Sprintf("%s/rsvswitchOverrideStpPol", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s",
+				"tDn": "%s"
```
Comment (AshuSoni-crest) [inline client/vmmVSwitchPolicyCont_service.go:133]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/vmmVSwitchPolicyCont_service.go [hunk 11]
```diff
@@ -0,0 +1,331 @@
...
+	dn := fmt.Sprintf("%s/rsvswitchOverrideStpPol", parentDn)
+	return sm.DeleteByDn(dn, "vmmRsVswitchOverrideStpPol")
+}
+func (sm *ServiceManager) CreateRelationvmmRsVswitchOverrideLldpIfPol(parentDn, LldpIfPolName string) error {
+	dn := fmt.Sprintf("%s/rsvswitchOverrideLldpIfPol", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s",
+				"tDn": "%s"
```
Comment (AshuSoni-crest) [inline client/vmmVSwitchPolicyCont_service.go:167]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/vmmVSwitchPolicyCont_service.go [hunk 12]
```diff
@@ -0,0 +1,331 @@
...
+	dn := fmt.Sprintf("%s/rsvswitchOverrideLldpIfPol", parentDn)
+	return sm.DeleteByDn(dn, "vmmRsVswitchOverrideLldpIfPol")
+}
+func (sm *ServiceManager) CreateRelationvmmRsVswitchOverrideMcpIfPol(parentDn, tnMcpIfPolName string) error {
+	dn := fmt.Sprintf("%s/rsvswitchOverrideMcpIfPol", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s",
+				"tDn": "%s"
```
Comment (AshuSoni-crest) [inline client/vmmVSwitchPolicyCont_service.go:201]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/vmmVSwitchPolicyCont_service.go [hunk 13]
```diff
@@ -0,0 +1,331 @@
...
+	dn := fmt.Sprintf("%s/rsvswitchOverrideMcpIfPol", parentDn)
+	return sm.DeleteByDn(dn, "vmmRsVswitchOverrideMcpIfPol")
+}
+func (sm *ServiceManager) CreateRelationvmmRsVswitchOverrideCdpIfPol(parentDn, CdpIfPolName string) error {
+	dn := fmt.Sprintf("%s/rsvswitchOverrideCdpIfPol", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s",
+				"tDn": "%s"
```
Comment (AshuSoni-crest) [inline client/vmmVSwitchPolicyCont_service.go:235]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/vmmVSwitchPolicyCont_service.go [hunk 14]
```diff
@@ -0,0 +1,331 @@
...
+	dn := fmt.Sprintf("%s/rsvswitchOverrideCdpIfPol", parentDn)
+	return sm.DeleteByDn(dn, "vmmRsVswitchOverrideCdpIfPol")
+}
+func (sm *ServiceManager) CreateRelationvmmRsVswitchOverrideLacpPol(parentDn, LacpLagPolName string) error {
+	dn := fmt.Sprintf("%s/rsvswitchOverrideLacpPol", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s",
+				"tDn": "%s"
```
Comment (AshuSoni-crest) [inline client/vmmVSwitchPolicyCont_service.go:269]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: client/vmmVSwitchPolicyCont_service.go [hunk 15]
```diff
@@ -0,0 +1,331 @@
...
+	return sm.DeleteByDn(dn, "vmmRsVswitchOverrideLacpPol")
+}
+
+func (sm *ServiceManager) CreateRelationvmmRsVswitchOverrideMtuPol(parentDn, tDn string) error {
+	dn := fmt.Sprintf("%s/rsvswitchOverrideMtuPol", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s",
+				"tDn": "%s"
```
Comment (AshuSoni-crest) [inline client/vmmVSwitchPolicyCont_service.go:304]: We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.

File: models/vmm_v_switch_policy_cont.go [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+
+	arr := make([]*VSwitchPolicyGroup, length)
+
+	for i := 0; i < length; i++ {
+
+		arr[i] = VSwitchPolicyGroupFromContainerList(cont, i)
+	}
+
+	return arr
+}
```
Comment (AshuSoni-crest) [inline models/vmm_v_switch_policy_cont.go:92]: reformatting is needed for this file

**Review by AshuSoni-crest — CHANGES_REQUESTED** [review]
Some changes needed. Added comments in the files

**Review by lhercot — APPROVED** [review]

**Pull request merged**